Add musig string and serde support

Ademan opened the pull request:
This addresses #796 to add serialization and deserialization for musig types that make sense to send around.

There might be an argument to also include every type with a `serialize()` but IMHO these are the types that make the most sense.

The test vectors come from https://github.com/Ademan/rust-secp256k1/tree/musig-serde-vectors because I wasn't sure if/what the licensing implications are of taking the vectors from bip-0327 (which is BSD licensed). However, they could be trivially dropped in if desired (convert to lowercase first).

The actual implementations are heavily "borrowed" from the serialization+deserialization of `schnorr::Signature`

Ademan commented:
It also occurred to me that the `serialize()` methods make constants like `MUSIG_KEYAGG_LEN` arguably part of the public interface, so maybe they should be re-exported in the musig module? (probably off-topic for this PR)

apoelstra commented:
Can you rebase on current `master` and also run `cargo +nightly fmt` on the result (or on every commit, whatever makes you happy)?

Otherwise this looks great. TIL that nonces and aggnonces are encoded as full uncompressed keys. It's a good idea that wouldn't have occurred to me. Agreed that we ought to expose these constants as part of the public API, but also agreed that it doesn't have to be part of this PR.

Ademan commented:
I think I'm already on top of current master, 4d36fefdddb118425bb9bcf611bb6e4dff306cfc right?

reformatted. Only the commit containing the test changed. I'll probably be out until this evening so if there's more feedback I probably won't be as responsive.

RE: nonces, I always forget the specifics of MuSig2 after a week so I could be misremembering, but aren't those two compressed points?

apoelstra commented:
Sorry, my local CI isn't passing on this (for the same reasons the real CI isn't passing). I'm going to merge #794 first and ask you to rebase.

Ademan commented:
NP, rebased and reran `cargo +nightly fmt` (no changes). I renamed my test in `musig.rs` to drop the `test_` prefix to match the rest, but everything else should be the same.

apoelstra reviewed (approved):
ACK 1d8b8621f8fc9b15310bf302bf9bc91eff85c82d; successfully ran local tests
